docs/source/app_developers_guide/docker.rst

```diff
@@ -0,0 +1,416 @@
+**************************
+Using Sawtooth with Docker
+**************************
+
+
+This document guides you through the process of setting up Hyperledger Sawtooth
+for application development using Docker Compose, and then introduces basic
+Sawtooth concepts necessary for application development.
+
+After following the steps in this document, you will be able to perform the
+following tasks:
+
+* run Sawtooth
+* submit transactions
+* view blocks, transactions and state with the CLI
+* start and stop validators and transaction processors
+
+You will then be prepared for the more advanced tutorials that guide you in
+performing app development tasks, such as implementing business logic with
+transaction families and writing clients which use Sawtooth's REST API.
+
+
+Overview Sawtooth Components
+============================
+
+A running Sawtooth network consists of the following applications or processes:
+
+.. image:: ../images/hyperledger_sawtooth_components.*
+   :width: 80%
+   :align: center
+   :alt: Sawtooth components
+
+This diagram represents a simple network with just two validators and two
+transaction processors. The second validator's transaction processors are not
+depicted.
+
+
+Installing Docker Engine and Docker Compose
+===========================================
+
+On Windows and macOS, install the latest version of `Docker Engine <https://docs.docker.com/engine/installation/>`_.
+
+  - On Windows and macOS, Docker Compose is installed automatically
+    when you install Docker Engine.
+
+On Linux, install both `Docker Engine <https://docs.docker.com/engine/installation/>`_ and `Docker Compose <https://docs.docker.com/compose/install/>`_.
+
+.. warning::
+
+  Note that the minimum version of Docker Engine necessary is 17.03.0-ce.
+  Linux distributions often ship with older versions of Docker.
+
+
+Installation
+============
+
+Download the Docker Compose File
+--------------------------------
+
+A Docker Compose file is provided which defines the process for constructing
+a Sawtooth environment, including the container images to download and the
+network settings.
```

Comment:
A Docker Compose file is provided which defines the process for constructing
a simple Sawtooth environment. This environment includes the following containers:

* A single validator using dev-mode consensus
* A REST-API connected to the validator
* Config, IntegerKey, and XO transaction processors
* A client container for running the CLI tools

The compose file also specifies the container images to download from Docker Hub and the
network settings needed for all the containers to communicate correctly.